### PR TITLE
Stdlib: Support evaluating len() at compile-time

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**

### CODE REVIEW

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -667,8 +667,8 @@ jobs:
         run: |
             git clone https://github.com/czgdp1807/stdlib.git
             cd stdlib
-            git checkout lf14
-            git checkout 6efd1982ba81a6bd8e21ea26173dafd42e8af8ab
+            git checkout lf15
```
Comment (Shaikh-Ubaid) [inline .github/workflows/CI.yml:670]: This branch `lf15` is checked out from `lf14`.

File: src/libasr/asr_utils.h [hunk 2]
```diff
@@ -1170,6 +1170,39 @@ static inline bool extract_value(ASR::expr_t* value_expr,
     return true;
 }
 
+static inline bool extract_value(ASR::expr_t* value_expr,
```
Comment (certik) [inline src/libasr/asr_utils.h:1173]: Is this only extracting string values? If so, let's name it:
```suggestion
static inline bool extract_string_value(ASR::expr_t* value_expr,
```

**Review by certik — APPROVED** [review]
I think this looks great. Left a minor comment, you can merge it after fixing it.

File: src/libasr/asr_utils.h (hunk 2 above)
Comment (Shaikh-Ubaid) [reply src/libasr/asr_utils.h]: Yes, updated.

**Pull request merged**